### PR TITLE
CI: add warning when not in pandas dev and running stubtest

### LINKED ISSUE #48260: CI: Improve warning when no pandas dev version installed when running stubtest
see https://github.com/pandas-dev/pandas/pull/48240#pullrequestreview-1086137999

### PR DESCRIPTION
**Pull request by ambujpawar**
- [x] closes #48260
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).

### CODE REVIEW

File: scripts/run_stubtest.py [hunk 1]
```diff
@@ -8,15 +8,26 @@
...
     if pd.compat.is_ci_environment():
         sys.exit(1)
     else:
         sys.exit(0)
 
+# GH 48260
+if "dev" not in pd_version:
+    warnings.warn(
+        f"stubtest may fail as {pd_version} is not a dev version. "
+        f"Please install a pandas dev version or see <https://github.com/pandas-dev"
```
Comment (mroeschke) [inline scripts/run_stubtest.py:26]: 1. Could you remove the `<` and `>`?
2. Could you link to `https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#validating-type-hints` instead?

Comment (ambujpawar) [reply scripts/run_stubtest.py]: Thanks for the review :)

Resolved the two issues you mentioned

**Comment by ambujpawar** [comment]
Thanks for the review. I resolved the comments you mentioned however, I noticed that the link you posted doesn't include the command to skip stubtest over [here](https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#validating-type-hints) compared to the one we have over [here](https://github.com/pandas-dev/pandas/blob/main/doc/source/development/contributing_codebase.rst#validating-type-hints)

Perhaps, there needs to be another task created for it? Or is it being done in another PR?

**Comment by mroeschke** [comment]
> Thanks for the review. I resolved the comments you mentioned however, I noticed that the link you posted doesn't include the command to skip stubtest over [here](https://pandas.pydata.org/pandas-docs/stable/development/contributing_codebase.html#validating-type-hints) compared to the one we have over [here](https://github.com/pandas-dev/pandas/blob/main/doc/source/development/contributing_codebase.rst#validating-type-hints)
> 
> Perhaps, there needs to be another task created for it? Or is it being done in another PR?

The doc updates in https://github.com/pandas-dev/pandas/pull/48240 will be reflected on the website soon. We have a CI job that updates the docs (will be applied to versions >= 1.5)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ambujpawar

**Pull request merged**